src/compaction_filter.rs

```diff
@@ -116,7 +114,9 @@ pub unsafe fn new_compaction_filter_raw(
         filter,
         name,
     );
-    crocksdb_ffi::crocksdb_compactionfilter_set_ignore_snapshots(filter, ignore_snapshots);
+    // In latest rocksdb `ignore_snapshots` will be always true.
+    // TODO: remove it after rocksdb is upgraded.
+    crocksdb_ffi::crocksdb_compactionfilter_set_ignore_snapshots(filter, true);
```

Comment:
This is not needed, since rocksdb 5.18 (tikv 3.x) already have the related change. Remove the C method as well.

Reply:
https://github.com/tikv/rocksdb/blob/tikv-3.x/include/rocksdb/compaction_filter.h#L176

Reply:
cases::test_compaction_filter::test_compaction_filter will fail on line 91. Seems by default snapshot won't be ignored?